rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/nested_array_with_constant_as_key.php.inc

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;
+
+use Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Source\GenericAnnotation;
+use Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Source\ConstantReference;
+
+/**
+ * @GenericAnnotation(
+ *     some={
+ *          ConstantReference::FIRST_NAME={
+ *              foo: {bar:{"baz"}}
+ *          },
+ *          ConstantReference::LAST_NAME
+ *     }
+ * )
+ */
+final class ArrayWithConstantAsKey
+{
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;
+
+use Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Source\GenericAnnotation;
+use Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Source\ConstantReference;
+
+#[GenericAnnotation(some: [ConstantReference::FIRST_NAME => ['foo' => ['bar' => ['baz']]], 0 => ConstantReference::LAST_NAME])]
```

Comment:
I would love to obtain this:
```suggestion
#[GenericAnnotation(some: [ConstantReference::FIRST_NAME => ['foo' => ['bar' => ['baz']]], ConstantReference::LAST_NAME])]
```
... but it seems basically impossible, since there's no way to say to PHP "this element has no key", it does in fact have one, just an autogenerated-due-to-append one.